metrics/client.go

```
@@ -592,16 +597,22 @@ func NewHawkularClient(p Parameters) (*Client, error) {
 		c.Transport = transport
 	}
 
+	var creds string
+	if p.Username != "" && p.Password != "" {
```

Only token or or only username/password can be allowed. Otherwise the client should throw an error when trying to initialize.

Having both would mean that the token overwrites the username/password authorization header. So this should check that the token isn't also set (or the token set should check that)